modules/heat_transfer/src/fvkernels/FVFunctorHeatConductionTimeDerivative.C

```diff
@@ -0,0 +1,45 @@
+//* This file is part of the MOOSE framework
+//* https://mooseframework.inl.gov
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FVFunctorHeatConductionTimeDerivative.h"
+
+registerMooseObject("HeatTransferApp", FVFunctorHeatConductionTimeDerivative);
+
+InputParameters
+FVFunctorHeatConductionTimeDerivative::validParams()
+{
+  InputParameters params = FVTimeKernel::validParams();
+  params.addClassDescription(
+      "AD Time derivative term $\\rho c_p \\frac{\\partial T}{\\partial t}$ of "
+      "the heat equation for quasi-constant specific heat $c_p$ and the density $\\rho$.");
+  params.addParam<MooseFunctorName>(
+      "specific_heat", "specific_heat", "Property name of the specific heat material property");
+  params.addParam<MooseFunctorName>(
+      "density", "density", "Property name of the density material property");
+  return params;
+}
+
+FVFunctorHeatConductionTimeDerivative::FVFunctorHeatConductionTimeDerivative(
+    const InputParameters & parameters)
+  : FVTimeKernel(parameters),
+    _specific_heat(getFunctor<ADReal>("specific_heat")),
+    _density(getFunctor<ADReal>("density"))
+{
+}
+
+ADReal
+FVFunctorHeatConductionTimeDerivative::computeQpResidual()
+{
+  // FV kernels dont use the quadrature
+  mooseAssert(_q_point.size() == 1, "Only 1 QP per element in FV");
```

Comment:
This is an overzealous assertion at this point. This would correctly evaluate the finite volume term even if there were multiple quadrature points in the quadrature rule unlike the qp-based term implementation

Reply:
Well this will be true after switching to use `FunctorTimeKernel` as the base class

Reply:
followed the change

Reply:
You're never supposed to use a qrule in an FV kernel

Reply:
Consequently, there is no loop over qrule qpoints in the base class like you say
```
void
FVElementalKernel::computeResidual()
{
  prepareVectorTag(_assembly, _var.number());
  _local_re(0) += MetaPhysicL::raw_value(computeQpResidual()) * _assembly.elemVolume();
  accumulateTaggedLocalResidual();
}
```

Reply:
right I agree with that